### PR TITLE
audit: Add object names in multi-delete API & trigger field

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
The trigger field shows who initiated the operation, for example: for
object deletion, is it an external request or the internal scanner.

Signed-off-by: Anis Elleuch <anis@min.io>

## Motivation and Context
Add trigger field to audit which shows what module triggered the operation (handlers, scanner,..)

## How to test this PR?
*) Setup an audit logger
*) Setup a bucket with lifecycle
*) Wait until lifecycle kicks in and watch trigger field

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: cmd/data-crawler.go [hunk 1]
```diff
@@ -44,7 +45,7 @@ import (
 const (
 	dataCrawlSleepPerFolder  = time.Millisecond // Time to wait between folders.
 	dataCrawlSleepDefMult    = 10.0             // Default multiplier for waits between operations.
-	dataCrawlStartDelay      = 5 * time.Minute  // Time to wait on startup and between cycles.
+	dataCrawlStartDelay      = 30 * time.Second // Time to wait on startup and between cycles.
```
Comment (harshavardhana) [inline cmd/data-crawler.go]:
```suggestion
	dataCrawlStartDelay      = 5 * time.Minute // Time to wait on startup and between cycles.
```

File: cmd/data-crawler.go [hunk 2]
```diff
@@ -755,6 +756,14 @@ func (i *crawlItem) applyActions(ctx context.Context, o ObjectLayer, meta action
...
 	}
 
+	entry := audit.NewEntry(globalDeploymentID)
+	entry.Trigger = "internal-scanner"
+	entry.API.Name = "DeleteObject"
+	entry.API.Bucket = i.bucket
+	entry.API.Object = i.objectPath()
+	ctx = logger.SetAuditEntry(ctx, &entry)
+	logger.AuditLog(ctx, nil, nil, nil)
+
```
Comment (harshavardhana) [inline cmd/data-crawler.go]: This can be a function like `auditLogLifecycle()`

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#171df314-a938-4a7b-9b6b-14f0610dc841) |
| mint-zoned.sh | :x: [more...](#af04c3df-bb82-4493-8955-d75314ed12ad) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#0818a2ad-378c-4656-ac2c-61a585e38b70) |
<a name="af04c3df-bb82-4493-8955-d75314ed12ad"></a>
### 12119-3a06023/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31265
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 970347d63175:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 25 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 21 seconds
(8/15) Running minio-dotnet tests ... done in 51 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 26 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 21 seconds
(11/15) Running minio-js tests ... done in 51 seconds
(12/15) Running minio-py tests ... FAILED in 3 minutes and 13 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-0fab5371-3798-4766-87a5-f887ca6e494d",
    "object_name": "fcc6579e-e8ac-47f0-8450-3f4cadf744ca"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-0fab5371-3798-4766-87a5-f887ca6e494d, request_id: 16783A819EB618BD, host_id: 95b3d08a-734c-4b9b-b167-4f81528ae229, bucket_name: minio-py-test-0fab5371-3798-4766-87a5-f887ca6e494d",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-0fab5371-3798-4766-87a5-f887ca6e494d, request_id: 16783A819EB618BD, host_id: 95b3d08a-734c-4b9b-b167-4f81528ae229, bucket_name: minio-py-test-0fab5371-3798-4766-87a5-f887ca6e494d\n",
  "duration": 6801
}
(12/15) Running s3cmd tests ... done in 22 seconds
(13/15) Running s3select tests ... done in 7 seconds
(14/15) Running security tests ... done in 1 seconds

Executed 14 out of 15 tests successfully.
```
<a name="0818a2ad-378c-4656-ac2c-61a585e38b70"></a>
### 12119-3a06023/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:32200
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 85f2d3e3559c:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2606,
  "function": "aws --endpoint-url http://minio-dev8.minio.io:32200 s3api copy-object --bucket awscli-mint-test-bucket-25138 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-25138/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 51 seconds
(7/15) Running minio-dotnet tests ... done in 36 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 39 seconds
{
  "args": {},
  "duration": 411,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 34 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 151,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-1md0eog, objectName = minio-java-test-7b4gk8-copy, resource = /minio-java-test-1md0eog/minio-java-test-7b4gk8-copy, requestId = 16783A296DDEF0C0, hostId = bb907ef1-2456-4100-a3a6-a87eb45ad09c)\nrequest={method=PUT, url=http://minio-dev8.minio.io:32200/minio-java-test-1md0eog/minio-java-test-7b4gk8-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-2c6ue26/minio-java-test-7b4gk8\nHost: minio-dev8.minio.io:32200\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210422T161335Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210422/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 416\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Thu, 22 Apr 2021 16:13:35 GMT\nServer: MinIO/DEVELOPMENT.2021-04-22T16-00-37Z\nVary: Origin\nX-Amz-Request-Id: 16783A296DDEF0C0\nX-Xss-Protection: 1; mode=block\nDate: Thu, 22 Apr 2021 16:13:35 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 1 minutes and 8 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 1 seconds
(10/15) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "3622",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-8230/s3cmd-test-object-32678 eacefcc4-caf5-4ab1-80ca-7d5af25fc589\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id eacefcc4-caf5-4ab1-80ca-7d5af25fc589 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
<a name="171df314-a938-4a7b-9b6b-14f0610dc841"></a>
### 12119-3a06023/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31223
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 158259b736f1:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 2 minutes and 11 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 53 seconds
(8/15) Running minio-dotnet tests ... done in 37 seconds
(9/15) Running minio-go tests ... done in 59 seconds
(10/15) Running minio-java tests ... done in 53 seconds
(11/15) Running minio-js tests ... done in 44 seconds
(12/15) Running minio-py tests ... FAILED in 2 minutes and 13 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-a6b42592-995a-4b00-a7f6-a3bf5e230e4f",
    "object_name": "544e3804-749b-49ef-a432-c7144b96291d"
  },
  "message": "(\"Connection broken: ConnectionResetError(104, 'Connection reset by peer')\", ConnectionResetError(104, 'Connection reset by peer'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/response.py\", line 438, in _error_catcher\n    yield\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/response.py\", line 519, in read\n    data = self._fp.read(amt) if not fp_closed else b\"\"\n  File \"/usr/lib/python3.6/http/client.py\", line 463, in read\n    n = self.readinto(b)\n  File \"/usr/lib/python3.6/http/client.py\", line 507, in readinto\n    n = self.fp.readinto(b)\n  File \"/usr/lib/python3.6/socket.py\", line 586, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1637, in test_thread_safe\n    raise exceptions[0]\n  File \"/mint/run/core/minio-py/tests.py\", line 1597, in get_object_and_check\n    _CLIENT.fget_object(bucket_name, object_name, local_file)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1085, in fget_object\n    for data in response.stream(amt=1024*1024):\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/response.py\", line 576, in stream\n    data = self.read(amt=amt, decode_content=decode_content)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/response.py\", line 541, in read\n    raise IncompleteRead(self._fp_bytes_read, self.length_remaining)\n  File \"/usr/lib/python3.6/contextlib.py\", line 99, in __exit__\n    self.gen.throw(type, value, traceback)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/response.py\", line 455, in _error_catcher\n    raise ProtocolError(\"Connection broken: %r\" % e, e)\nurllib3.exceptions.ProtocolError: (\"Connection broken: ConnectionResetError(104, 'Connection reset by peer')\", ConnectionResetError(104, 'Connection reset by peer'))\n",
  "duration": 3619
}
(12/15) Running s3cmd tests ... done in 1 minutes and 4 seconds
(13/15) Running s3select tests ... done in 4 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/data-crawler.go (hunk 1 above)
Comment (vadmeste) [reply cmd/data-crawler.go]: Fixed !! Thanks for catching this

File: cmd/data-crawler.go (hunk 2 above)
Comment (vadmeste) [reply cmd/data-crawler.go]: > This can be a function like `auditLogLifecycle()`

Fixed

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#9abf2560-d17c-4482-bb19-ba3fc9284c9f) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#48c1eddf-f97a-4f24-aa9f-df9a83887d73) |
<a name="9abf2560-d17c-4482-bb19-ba3fc9284c9f"></a>
### 12119-0325773/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:32763
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp daea2648f296:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 2 minutes and 18 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 2 seconds
(8/15) Running minio-dotnet tests ... done in 56 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 12 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 19 seconds
(11/15) Running minio-js tests ... done in 50 seconds
(12/15) Running minio-py tests ... FAILED in 3 minutes and 14 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-0abb9eb8-ca46-457b-9374-8437f4a1bf2e",
    "object_name": "89167a85-4d89-4b9a-8e76-1b368d494e47"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-0abb9eb8-ca46-457b-9374-8437f4a1bf2e, request_id: 1678484271EF2205, host_id: dd570250-4922-4319-b21e-2894bc874fe0, bucket_name: minio-py-test-0abb9eb8-ca46-457b-9374-8437f4a1bf2e",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-0abb9eb8-ca46-457b-9374-8437f4a1bf2e, request_id: 1678484271EF2205, host_id: dd570250-4922-4319-b21e-2894bc874fe0, bucket_name: minio-py-test-0abb9eb8-ca46-457b-9374-8437f4a1bf2e\n",
  "duration": 6255
}
(12/15) Running s3cmd tests ... done in 20 seconds
(13/15) Running s3select tests ... done in 8 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="48c1eddf-f97a-4f24-aa9f-df9a83887d73"></a>
### 12119-0325773/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30889
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ebe6bdb3998d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2837,
  "function": "aws --endpoint-url http://minio-c3.minio.io:30889 s3api copy-object --bucket awscli-mint-test-bucket-14612 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-14612/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 51 seconds
(7/15) Running minio-dotnet tests ... done in 43 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 34 seconds
{
  "args": {},
  "duration": 513,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 1 minutes and 27 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 719,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-2lpfg5u, objectName = minio-java-test-2k60ihs-copy, resource = /minio-java-test-2lpfg5u/minio-java-test-2k60ihs-copy, requestId = 167847F23111B5A4, hostId = 5905e0e3-2972-4682-b10e-e87ee90e42fe)\nrequest={method=PUT, url=http://minio-c3.minio.io:30889/minio-java-test-2lpfg5u/minio-java-test-2k60ihs-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-39qb00g/minio-java-test-2k60ihs\nHost: minio-c3.minio.io:30889\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210422T202611Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210422/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Thu, 22 Apr 2021 20:26:11 GMT\nServer: MinIO/DEVELOPMENT.2021-04-22T20-13-23Z\nVary: Origin\nX-Amz-Request-Id: 167847F23111B5A4\nX-Xss-Protection: 1; mode=block\nDate: Thu, 22 Apr 2021 20:26:11 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 52 seconds
(9/15) Running minio-py tests ... done in 3 minutes and 4 seconds
(10/15) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "3620",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-2222/s3cmd-test-object-24484 a4c2f888-8fb7-42b5-a961-11ad7f962537\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id a4c2f888-8fb7-42b5-a961-11ad7f962537 put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 1 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by vadmeste** [comment]
@harshavardhana can you merge this if no other comments? We need to build hotfix binary.

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**